cellxgene_schema_cli/cellxgene_schema/schema_definitions/schema_definition.yaml

```diff
@@ -226,31 +228,34 @@ components:
                             exceptions:
                                 - unknown
                             forbidden:
-                                - HANCESTRO:0002
-                                - HANCESTRO:0003
-                                - HANCESTRO:0004
-                                - HANCESTRO:0018
-                                - HANCESTRO:0029
-                                - HANCESTRO:0030
-                                - HANCESTRO:0031
-                                - HANCESTRO:0032
-                                - HANCESTRO:0033
-                                - HANCESTRO:0034
-                                - HANCESTRO:0290
-                                - HANCESTRO:0304
-                                - HANCESTRO:0323
-                                - HANCESTRO:0324
-                                - HANCESTRO:0551
-                                - HANCESTRO:0554
-                                - HANCESTRO:0555
-                                - HANCESTRO:0557
-                                - HANCESTRO:0558
-                                - HANCESTRO:0559
-                                - HANCESTRO:0560
-                                - HANCESTRO:0561
-                                - HANCESTRO:0564
-                                - HANCESTRO:0565
-                                - HANCESTRO:0566
+                                terms:
+                                    - HANCESTRO:0003
+                                    - HANCESTRO:0004
+                                    - HANCESTRO:0018
+                                    - HANCESTRO:0029
+                                    - HANCESTRO:0030
+                                    - HANCESTRO:0031
+                                    - HANCESTRO:0032
+                                    - HANCESTRO:0033
+                                    - HANCESTRO:0034
+                                    - HANCESTRO:0290
+                                    - HANCESTRO:0323
+                                    - HANCESTRO:0324
+                                    - HANCESTRO:0551
+                                    - HANCESTRO:0554
+                                    - HANCESTRO:0555
+                                    - HANCESTRO:0557
+                                    - HANCESTRO:0558
+                                    - HANCESTRO:0559
+                                    - HANCESTRO:0560
+                                    - HANCESTRO:0561
+                                    - HANCESTRO:0564
+                                    - HANCESTRO:0565
+                                    - HANCESTRO:0566
+                                ancestors_inclusive:    # these terms AND their children are forbidden
```

Comment:
we use 'ancestor' / 'ancestor_inclusive' elsewhere in the schema definition to refer to listing out parent terms, so I'd like to stay consistent. I don't mind switching over everywhere, though I read it as we are listing out the forbidden ancestors (and the '_inclusive' means, in this case, we are counting the term as an ancestor of itself, but we may not always want to do that). Maybe we can re-word the comment instead, but let me know if you think its more clear using 'descendants' instead.

Reply:
- it would be easier to to understand if it changed to parent/child(which is more commonly understood programmatically) vs ancestors/descendants. 
- `ancestors_inclusive` sounds like this term and it's ancestors are forbidden.

Is it invalid for a term to be in both `ancestors_inclusive` and `term`? We may need a tests case to know the behavior and possible a schema check to avoid accidentally producing an invalid schema.

Reply:
could we drop the `ancestor` part?

Reply:
We control the yaml format, so we could re-work it so that terms can be in both term and ancestor. But as written, yes, it would be invalid for a term to be in both. 

We use ancestor/descendant because its in reference to ontologies, which use that terminology. 

We definitely need to keep either ancestor/descendant or parent/child, because we need a way to differentiate between 

1) terms that are forbidden but its children are allowed
2) terms that are forbidden AND its children are forbidden
3) terms that are allowed BUT its children are forbidden 

While not a requirement, it would be nice for that naming convention to be consistent with the opposite use-case which is already using ancestor/ancestor_inclusive, which is whitelists of:

1) terms that are allowed but its children are not
2) terms that are allowed AND its children are allowed
3) terms that are not allowed BUT its children are allowed

Reply:
What do you think of simplifying it to forbidden terms and forbidden ancestors? and if its both, the term should just be listed in both.

Reply:
it would require a refactor of the whitelist version, which already uses ancestors_inclusive, but that's fine if its clearer overall

Reply:
That would make more sense.

Reply:
the refactor to make that happen + make the allowlist consistent with that pattern as well was more involved than I anticipated. Pushed it, let me know what you think. If the review around this becomes any more complex I may suggest reverting the change and taking care of it in a separate PR instead